github3/session.py

```diff
@@ -125,8 +125,9 @@ def token_auth(self, token):
         self.headers.update({
             'Authorization': 'token {0}'.format(token)
             })
-        # Unset username/password so we stop sending them
-        self.auth = None
+        # Requests requires a callable for 'auth' otherwise it
+        # will attempt to use any auth in .netrc
+        self.auth = lambda x: x
```

Comment:
We could do this or we could set `trust_env = False`. Setting a "null auth"  might affect our decorators that introspect the auth on a session.

Reply:
Setting `trust_env = False` breaks use of http_proxy/https_proxy/no_proxy env settings, and would require then explicit handling of proxy env vars from this library down to python requests.

Reply:
I wonder if it would make more sense to to treat our authentication better.

Should we have a proper auth handler for this, e.g.,

```py
class TokenAuth(requests.auth.AuthBase):
    def __init__(self, token):
        self.token = token

    def __call__(self, request):
        request.headers['Authorization'] = 'token {}'.format(token)
```

That could negate this entirely.